proj1a/LinkedListDeque.java

```diff
@@ -0,0 +1,145 @@
+public class LinkedListDeque<T> implements Deque<T> {
+    private class LLDNode {
+        LLDNode prev;
+        T item;
+        LLDNode next;
+
+        public LLDNode(LLDNode front, T n, LLDNode back) {
+            prev = front;
+            item = n;
+            next = back;
+        }
+    }
+
+    private LLDNode sentinel;
+    private int size;
+
+    /**create empty linked list deque*/
+    public LinkedListDeque() {
+        sentinel = new LLDNode (null, null, null);
+        sentinel.prev = sentinel;
+        sentinel.next = sentinel;
+        size = 0;
+    }
+
+    /**add item to front of the deque*/
+    @Override
+    public void addFirst(T item) {
```

Comment:
You could make this implementation [thread safe](https://www.journaldev.com/1061/thread-safety-in-java) by adding `synchronized` keyword:

```
    public synchronized void addFirst(T item) {
```